File: pdu/python/hako_srv_msgs/pdu_cdr_conv_GetEventRequestPacket.py
from .pdu_pytype_GetEventRequestPacket import GetEventRequestPacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_GetEventRequest import *
from ..hako_srv_msgs.pdu_cdr_conv_ServiceRequestHeader import *



def py_to_cdr_body_GetEventRequestPacket(writer: CdrWriter, src: GetEventRequestPacket):
    py_to_cdr_body_ServiceRequestHeader(writer, src.header)
    py_to_cdr_body_GetEventRequest(writer, src.body)


def cdr_body_to_py_GetEventRequestPacket(reader: CdrReader, dst: GetEventRequestPacket):
    cdr_body_to_py_ServiceRequestHeader(reader, dst.header)
    cdr_body_to_py_GetEventRequest(reader, dst.body)
    return dst


def py_to_cdr_GetEventRequestPacket(src: GetEventRequestPacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_GetEventRequestPacket(writer, src)
    return writer.bytes()


def cdr_to_py_GetEventRequestPacket(cdr_payload) -> GetEventRequestPacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = GetEventRequestPacket()
    return cdr_body_to_py_GetEventRequestPacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_GetEventRequestPacket.js
import { GetEventRequestPacket } from './pdu_jstype_GetEventRequestPacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { GetEventRequest } from '../hako_srv_msgs/pdu_jstype_GetEventRequest.js';
import { PduGetEventRequestConverter } from '../hako_srv_msgs/pdu_cdr_conv_GetEventRequest.js';
import { ServiceRequestHeader } from '../hako_srv_msgs/pdu_jstype_ServiceRequestHeader.js';
import { PduServiceRequestHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceRequestHeader.js';


export class PduGetEventRequestPacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { GetEventRequestPacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceRequestHeaderConverter.to_cdr_body(writer, src.header);
        PduGetEventRequestConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { GetEventRequestPacket } dst
     * @returns { GetEventRequestPacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceRequestHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduGetEventRequestConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { GetEventRequestPacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { GetEventRequestPacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new GetEventRequestPacket());
    }
}
